ansible_collections/arista/avd/roles/eos_designs/templates/facts/switch/switch.j2

```diff
@@ -147,22 +153,58 @@ switch:
 {# switch.router_id #}
   router_id: {% include switch.ip_addressing.router_id %}
 
+{%     if underlay_ipv6 is arista.avd.defined(true) %}
+{%         if switch_data.combined.loopback_ipv6_pool is arista.avd.defined(fail_action='error',var_name='loopback_ipv6_pool') %}
+{# IPv6 related facts #}
+{# switch.loopback_ipv6_pool #}
+{%             set loopback_ipv6_pool = switch_data.combined.loopback_ipv6_pool %}
+  loopback_ipv6_pool: {{ loopback_ipv4_pool }}
+
+{# switch.loopback_ipv4_offset #}
+{%             set loopback_ipv6_offset = switch_data.combined.loopback_ipv6_offset | arista.avd.default(0) %}
+  loopback_ipv6_offset: {{ loopback_ipv6_offset }}
+
+{# switch.loopback_ipv4_description #}
+  loopback_ipv6_description: {{ switch_data.combined.loopback_ipv6_description | arista.avd.default(overlay_loopback_description) }}
+
+{# switch.ipv6_router_id #}
+  ipv6_router_id: {% include switch.ip_addressing.ipv6_router_id %}
+{%         endif %}
+{%     endif %}
+
+{# switch.evpn_role #}
+{%     set switch_evpn_role = switch_data.combined.evpn_role | arista.avd.default(
+                          switch.default_evpn_role, 'none') %}
+  evpn_role: {{ switch_evpn_role }}
+
+{# switch.mpls_overlay_role #}
+{%     set switch_mpls_overlay_role = switch_data.combined.mpls_overlay_role | arista.avd.default(
+                          switch.default_mpls_overlay_role, 'none') %}
```

Comment:
remove the last 'none' fallback. it is already covered in the node_type logic. Same for evpn

Reply:
Fixed.